tests/test_fmi.py

```diff
@@ -36,6 +39,16 @@
 
 file_path = os.path.dirname(os.path.abspath(__file__))
 
+
+def _helper_unzipped_fmu_exception_invalid_dir(fmu_loader):
+    """ Verify that we get an exception if unzipped FMU does not contain modelDescription.xml, which it should according to the FMI specification.
+        The input argument is any of the FMU interfaces FMUModelME1, FMUModelME2, FMUModelCS1, FMUModelCS2 and load_fmu from pyfmi.fmi.
+    """
+    err_msg = "Specified fmu path '.*\\' needs to contain a modelDescription.xml according to the FMI specification"
+    with tempfile.TemporaryDirectory() as temp_dir:
+        with np.testing.assert_raises_regex(FMUException, err_msg):
```

Comment:
Why a with statement?

Reply:
I used a with-statement since TemporaryDirectory() can be used as a context manager. Basically we don't need to invoke mkdir/rmdir by ourselves.
If you are referring to the exception, it is the preferred way to test exceptions (as far as I know). It will also throw an exception if no exception was thrown.

Reply:
Hmm ok. I was referring to the with statement on "with np.testing.assert_raises_regex" I've not seen this before and don't think we have it anywhere else in the code base. Why is not "np.testing.assert_raises_regex(FMUException, ...., err_msg)" enough?

Reply:
As far as I understand, this is the only difference according to their documentation:

> An optional keyword argument 'msg' can be provided when assert_raises
> is used as a context object.

Other than that I just think context managers are better since they allocate and release resources precisely within that block.

Reply:
Hmm ok. Still I must say that I think it becomes harder to read what happens when context managers are introduced.